packages/frontend/src/components/ProjectAccess/ProjectAccess.tsx

```diff
@@ -43,23 +80,42 @@ const ProjectAccess: FC<ProjectAccessProps> = ({
         }),
     );
 
+    // Reset page when search changes
+    useEffect(() => {
+        setPage(1);
+    }, [search]);
+
     const filteredUsers = useMemo(() => {
         if (search && usersWithProjectRole) {
             return new Fuse(usersWithProjectRole, {
-                keys: ['firstName', 'lastName', 'email', 'finalRole'],
+                keys: ['firstName', 'lastName', 'email', 'role', 'projectRole'],
                 ignoreLocation: true,
                 threshold: 0.3,
             })
                 .search(search)
-                .map((result) => ({
-                    ...result.item,
-                    inheritedRole: inheritedRoles?.[result.item.userUuid],
-                }));
+                .map((result) => result.item);
         }
         return usersWithProjectRole;
-    }, [usersWithProjectRole, search, inheritedRoles]);
+    }, [usersWithProjectRole, search]);
+
+    // Pagination logic
+    const paginatedUsers = useMemo(() => {
+        if (!filteredUsers) return [];
+        const startIndex = (page - 1) * DEFAULT_PAGE_SIZE;
+        const endIndex = startIndex + DEFAULT_PAGE_SIZE;
+        return filteredUsers.slice(startIndex, endIndex);
+    }, [filteredUsers, page]);
+
+    const totalPages = useMemo(() => {
```

Comment:
Do we have a criteria for when to leverage useMemo? I ask because it doesn't come for free and cases like this one are often more expensive than simply creating `paginatedUsers` with each render. I'm definitely not dogmatic, just curious.